### DISCUSSION TITLE
[Feature] PHP extension supports link tracking sampling rate configuration

### DISCUSSION THREAD

**Discussion by elricli** [apache/skywalking#12044] [General]
### Description

Like:
- `skywalking.sample_n_per_3_secs` in the [SkyAPM/SkyAPM-php-sdk v4](https://github.com/SkyAPM/SkyAPM-php-sdk/blob/205bfc1c1d15c1fe1f45821a0948562df7bbdec2/skywalking.c#L75) 
- `agent.sampler` in the [skywalking-go](https://github.com/apache/skywalking-go/blob/642606007162e9b6f5c6827e4903994f6364d8fe/tools/go-agent/config/agent.default.yaml#L26) 

### Use case

Added a php.ini configuration: `skywalking_agent.sampler` (Sampling rate of tracing data, which is a floating-point value that must be between 0 and 1.)

### Related issues

#12043

**Comment by jmjoy** [comment]
I haven't looked at the Go implementation, but I have a question:

Currently, various languages agent don't uniformly apply the first sample field of the `sw8` header of the Cross Process Propagation Header. If the sampling rate is turned on and the Context is not sampled, the following problems may occur:

1. If Context is a user portal, the called Context mistakenly becomes the user portal.
2. If Context is called, the trace will be missing or broken into two segments.

Because of these reasons, I don’t know how to implement sampler better.

**Reply by wu-sheng** [reply]
@jmjoy The sampler provided above is not about the `sw8` header compliance, that field is not used. 
The feature mentioned above is when you can't read header, which means this is the first service in the whole distributed chain, you would use that sampler mechanism. Otherwise, once the header exists, the agent does force sampling by ignoring this sampling mechanism.

Is this clear to you?

**Reply by jmjoy** [reply]
> 1. If Context is a user portal, the called Context mistakenly becomes the user portal.

So this situation is possible?

**Reply by wu-sheng** [reply]
> > 1. If Context is a user portal, the called Context mistakenly becomes the user portal.
> 
> 
> 
> So this situation is possible?
> 
> 

There is not a context called that.
This is not the agent side concern.

**Comment by heyanlong** [comment]
Based as my know, the feature just works first service, if current server can't read `sw8` header then it's first service
